src/wallet/rpcwallet.cpp

```diff
@@ -3013,9 +3013,11 @@ UniValue listunspent(const JSONRPCRequest& request)
 
     UniValue results(UniValue::VARR);
     std::vector<COutput> vecOutputs;
-    LOCK2(cs_main, pwallet->cs_wallet);
+    {
+        LOCK2(cs_main, pwallet->cs_wallet);
+        pwallet->AvailableCoins(vecOutputs, !include_unsafe, nullptr, nMinimumAmount, nMaximumAmount, nMinimumSumAmount, nMaximumCount, nMinDepth, nMaxDepth);
+    }
```

Comment:
I thought taking a copy of the address book here to avoid locking the wallet bellow. Thoughts?

Reply:
As it contains an entry for every key, the address book can be quite a big data structure. I don't think making a copy is a good trade-off.

Reply:
Agree.